### PR TITLE
scry: does not build with latest crystal

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Use 0.35 until it gains support for 0.36+

Cc: @NixOS/crystal-lang 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
Upstream: https://github.com/crystal-lang-tools/scry/issues/179

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -9569,7 +9569,7 @@ in
 
   icr = callPackage ../development/tools/icr { };
 
-  scry = callPackage ../development/tools/scry {};
+  scry = callPackage ../development/tools/scry { crystal = crystal_0_35; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:9572]: You need to change the input in https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/scry/default.nix#L1 to crystal.

Comment (peterhoeg) [reply pkgs/top-level/all-packages.nix:9572]: Do you have a reference to where we've decided to use that approach as opposed to just overriding the input? IMHO is less messy to do it this way although it is of course less clear for someone who is just looking at the derivation itself in isolation to figure out what is going on.

Comment (peterhoeg) [reply pkgs/top-level/all-packages.nix:9572]: Ahhh, I had my own patched version of scry here.

Comment (peterhoeg) [reply pkgs/top-level/all-packages.nix:9572]: That's why it was building for me.

**Review by Br1ght0ne — APPROVED** [review]
Diff LGTM.

**Pull request merged**